### PR TITLE
Add hooks for entities hiding in blocks

### PR DESCRIPTION
**Pull request by diesieben07**
Open for debate, as there is demand for it but I'm not sure if this is the best way to do it.

### CODE REVIEW

**Comment by LexManos** [comment]
Its a bit to specific for 'silverfish' what if other mods add other 'hide in block' type mobs?

**Comment by ThePsionic** [comment]
Wouldn't those modders be responsible for making their own hooks, then?

**Comment by Eragonn1490** [comment]
The point of forge is to make it easier for all modders, so I assume lex was telling not suggesting

**Comment by ThePsionic** [comment]
True, but if a mod adds their own block-entering entities, shouldn't the modders themselves be responsible for making the hooks for those events

**Comment by r4j4h** [comment]
Perhaps, but couldn't Forge provide a generic hide-in-block-thing-entered and hide-in-block-thing-exited event. Then both silverfish and future mods can use that generic hook and it all makes sense semantically.

Do you really think having 6 different entry/exit hooks will be performant? Especially if each is implemented differently by people of different skill levels? That scares me personally.

**Comment by ThePsionic** [comment]
Fair enough. I retract my point.

**Comment by diesieben07** [comment]
Seems like I got work to do.

**Comment by cpw** [comment]
Yup, I think you're analysis is correct, don't type the event just for silverfish...

**Comment by r4j4h** [comment]
I am excited to see the end result, this will be a really cool feature :]

**Comment by diesieben07** [comment]
Better? :)

File: patches/minecraft/net/minecraft/block/BlockSilverfish.java.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- ../src-base/minecraft/net/minecraft/block/BlockSilverfish.java
++++ ../src-work/minecraft/net/minecraft/block/BlockSilverfish.java
+@@ -36,6 +36,7 @@
+         return 0;
+     }
+ 
++    @Deprecated // use Block.canContainSilverfish
```
Comment (simon816) [inline patches/minecraft/net/minecraft/block/BlockSilverfish.java.patch]: Outdated comment.

Comment (diesieben07) [inline patches/minecraft/net/minecraft/block/BlockSilverfish.java.patch]: Yeah. I'll fix it, thanks.

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Comment by diesieben07** [comment]
Is there still demand for this? If so I'll update it for 1.9.4.

**Comment by r4j4h** [comment]
I don't honestly have a demand for it but I would like to see it because blessed hook points are always good, if the maintainers say they are willing to accept the merge if the conflicts were resolved.

**Pull request closed without merging**